File: at_coder/C/abc_008/01.go
package main

import (
	"fmt"
)

func main() {
	var n int
	fmt.Scan(&n)
	a := make([]int, n)

	for i := 0; i < n; i++ {
		fmt.Scan(&a[i])
	}

	var ans = 0. //float64(n)
	for i := 0; i < n; i++ {
		var cnt = 0
		for j := 0; j < n; j++ {
			if a[i]%a[j] == 0 {
				cnt++
			}
		}
		ans += map[bool]float64{true: 0.5, false: float64(cnt+1) / float64(2*cnt)}[cnt%2 == 0]
	}
	fmt.Println(ans)
}

File: at_coder/C/abc_008/01.ts
function float2int (value) {
    return value | 0;
}

"use strict";
function main(input) {
    const args = input.split('\n');
    const n = parseInt(args[0], 10);
    let a = [];
    for (let i = 1; i <= n; i++) {
        a.push(args[i]);
    }
    let ans = 0.0;
    for (let i = 0; i < n; i++) {
        let cnt = 0;
        for (let j = 0; j < n; j++) {
            if (a[j] % a[i] === 0) cnt++;
        }
        ans += float2int((cnt + 1) / 2) / cnt;
    }
    console.log(ans);
}

main(require('fs').readFileSync('/dev/stdin', 'utf8'));
